Remove docs dependency on deprecated rhelimage storage account and add RHEL 10 docs

sdherr opened the pull request:
We (the RHUI team) are in the process of deprecating this `rhelimage` storage account. This PR updates our public-facing docs to replace the references to it with spelled-out configuration files, which makes for more complete documentation anyway. It also updates documentation for the release of RHEL 10 and removes references to out-of-support versions of RHEL.

v-regandowner commented:
@ju-shim - Can you review the proposed changes? 
	
IMPORTANT: When the changes are ready for publication, adding a `#sign-off` comment is the best way to signal that the PR is ready for the review team to merge. 
	
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

sdherr commented:
#sign-off

v-dirichards commented:
@ju-shim @iamwilliew @mattmcinnes 

Could you review or reassign this pull request? Thanks!

sdherr commented:
This PR has been pending for so long that I have needed to make additional changes: I have added docs for RHEL 10 and removed docs related to deprecated versions of RHEL.
@ju-shim  @iamwillbar  @mattmcinnes Please review. It is ready from my perspective.

v-dirichards commented:
@ju-shim @iamwilliew @mattmcinnes

Could you review this proposed update to your article and enter `#sign-off` in a comment if it's ready to merge?

Thanks.

v-dirichards commented on articles/virtual-machines/workloads/redhat/redhat-in-place-upgrade.md:
```diff
@@ -27,20 +27,18 @@ ms.author: jushiman
 During an in-place upgrade, the earlier RHEL OS major version will be replaced with the new RHEL OS major version without removing the earlier version first. The installed applications and utilities, along with the configurations and preferences, are incorporated into the new version.
 
+## Upgrade RHEL VMs
+In general Red Hat only supports upgrading from version to the next; you cannot skip versions. See the Red Hat's documentation on in-place upgrades:
```
```suggestion
In general, Red Hat only supports upgrading from version to the next; you cannot skip versions. See the Red Hat's documentation on in-place upgrades:
```
punctuation
